### PR TITLE
Add docs for GRPCRoute

### PR DESCRIPTION
**Pull request by arkodg**
Fixes: https://github.com/envoyproxy/gateway/issues/642

Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

**Comment by Xunzhuo** [comment]
Is it a blocker for v0.3.0-rc1?

**Comment by arkodg** [comment]
> Is it a blocker for v0.3.0-rc1?

no, not a must have for 0.3.0-rc1

**Comment by HoangTheQuyen** [comment]
spec:
  parentRefs:
    - name: eg
  I think it should be:
  spec:
    parentRefs:
      - name: example-gateway

**Comment by HoangTheQuyen** [comment]
spec:
  parentRefs:
    - name: eg
 I think name should be: example-gateway

**Comment by arkodg** [comment]
thanks for catching that @HoangTheQuyen , updated the PR

File: docs/latest/user/grpc-routing.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Testing the Configuration
+
+Before testing GRPC routing to the `yages` backend, get the Gateway's address.
+
+```shell
+export GATEWAY_HOST=$(kubectl get gateway/example-gateway -o jsonpath='{.status.addresses[0].value}')
+```
+
+Test GRPC routing to the `yages` backend using the `grpcurl` command.
```
Comment (Xunzhuo) [inline docs/latest/user/grpc-routing.md:73]: https://github.com/fullstorydev/grpcurl 

Can we add a ref to grpcurl link?

**Review by Xunzhuo — COMMENTED** [review]
nit

File: docs/latest/user/grpc-routing.md (hunk 1 above)
Comment (arkodg) [reply docs/latest/user/grpc-routing.md]: ack

**Comment by danehans** [comment]
@arkodg thanks for the PR. I have confirmed the steps in the docs. Please send a follow-up PR that adds this doc to the [user docs index page](https://github.com/envoyproxy/gateway/blob/main/docs/latest/user_docs.rst).

**Review by danehans — APPROVED** [review]

**Comment by arkodg** [comment]
> @arkodg thanks for the PR. I have confirmed the steps in the docs. Please send a follow-up PR that adds this doc to the [user docs index page](https://github.com/envoyproxy/gateway/blob/main/docs/latest/user_docs.rst).

ah thanks for reminding me to take of this, handled it in this PR itself, ptal

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**